src/components/Suspense.astro

```diff
@@ -1,7 +1,72 @@
 ---
-const idx = Astro.locals.suspend(Astro.slots.render("default"));
+type PendingThenable<T> = { status: "pending" };
+type FulfilledThenable<T> = { status: "fulfilled"; value: T };
+type RejectedThenable<T> = { status: "rejected"; reason: unknown };
+
+type Thenable<T> = Promise<T> &
+  (PendingThenable<T> | FulfilledThenable<T> | RejectedThenable<T>);
+
+function instrument<T>(promise: Promise<T>): Thenable<T> {
```

Comment:
nit: I'm not familiar with the term "instrument." Could we change this to something like "asTrackedPromise" since it applies a `status` and `reason`?

Reply:
i was thinking "instrument" as "add instrumentation", but i'm not very attached to that

re: `asTrackedPromise` -- i guess that could be a bit misleading, bc this patches the original promise (only returning a new thing for typescript convenience). that's why i like "instrument" bc it implies mutation

Reply:
Gotcha. Maybe a verb like `trackPromise` would be better. You're right that `as` does not imply mutability. Also may just be me, but haven't heard "instrumentation" used often either 😅